app/src/main/java/waiter/Communicator/Reporter.java

```diff
@@ -0,0 +1,8 @@
+package waiter.Communicator;
+
+import java.io.IOException;
+
+public interface Reporter {
```

Comment:
I noticed there are quite a number of interfaces now and it's getting a bit difficult keeping track of what they are vs an actual class. I wonder if there are some organizationally stuff we can do to make this more manageable.

Reply:
Hmmm, I have put each interface in a package with their associated mock (in a mock package within said package) and classes, so for example, the Reporter interface lives in a Communicator package with the Communicator class and a mock package that contains the mockCommunicator, both of which implement this interface. No class or mock implements an interface that they are not packaged with. Do you think having an interface package (like the mock) would be a good idea?

Reply:
I'm not sure. I've seen file organization where actually classes live in an implementation directory and interfaces live in another.

Reply:
I was also thinking about a file structure that looks like this:
```
---Reportable/ (interface name)
     -(interface goes here)
     --Implementation/
         -(classes go here)
     --Mocks/
        -(mocks go here)
```

Reply:
If we implement the noun -> class, adjective -> interface, would you still recommend an organizational change like this?

Reply:
Hmmm, I do like this but I also have a question about it. Do you think the interface should be the directories name? Just because I feel like the implementations name is more recognizable if that makes sense. Actually the more that I think about it the more this makes sense over having the implementation name be the directory as there is only one interface to this directory but there could be multiple classes which you wouldn't want just floating around in the initial directory. I really like this organizational structure here 😄

Reply:
Yea I'm not super sure about this structure either and I think you have a point about class name as directory name being more clear. I thought about this structure when I think about EchoProtocol and HTTPProtocol that will both implement the Protocol interface. In that case it might makes sense to put them both under `Protocol` which is the name for the interface. Because if you create separate directory for these classes, that implies that the interface will need to live alone somewhere. The thing I'm not super sure is mixing that with classes that don't implement any interface, then you end up having directories with interface names AND class names which can also be confusing.

Reply:
Hmmm, would it be ok to keep it like this for now and then create a tracking ticket and address this if I finish the other tickets this sprint or next sprint? I think there is definitely a better way to organize this (some form of what we have both suggested/discussed) and want to implement it but don't want it to prevent me from continuing on things this sprint since I feel its not incredibly blocking. What are your thoughts? 😄

Reply:
Yea definitely not blocking! I think updating the interface name to be adjectives is a good first step.

Reply:
A ticket was created to track the apps interface/class reorganization